File: frontend/src/types/problem.ts
import type { Category } from "./category";
import type { Equipment } from "./equipment";

export interface Problem {
  idProblem: number;
  title: string;
  description: string;
  voteCount?: number;
  category: Category;
  equipment?: Equipment;
  createdAt?: string;

  user?: {
    idUsers: number;
    username: string;
    email?: string;
    solutionCount?: number;
    badge?: string;
  };
}

export interface ProblemCreate {
  title: string;
  description: string;
  idCategory: number;
  idEquipment?: number;
}

File: backend/src/main/java/com/solvhub/model/Problem.java
package com.solvhub.model;

import jakarta.persistence.*;

import lombok.*;

import java.time.Instant;

@Entity
@Table(name = "problem")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Problem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer idProblem;

    @Column(nullable = false, length = 255)
    private String title;

    @Column(length = 1000)
    private String description;

    @ManyToOne
    @JoinColumn(name = "id_category", nullable = false)
    private Category category;

    @Column(nullable = false, updatable = false)
    private Instant createdAt;

    @PrePersist
    public void prePersist() {
        this.createdAt = Instant.now();
    }

    // 🔗 relation avec User
    @ManyToOne
    @JoinColumn(name = "id_users", nullable = true)
    private User user;

    @ManyToOne
    @JoinColumn(name = "id_equipment", nullable = true)
    private Equipment equipment;
}